experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyTree.ts

```diff
@@ -408,20 +410,34 @@ export class LazyMap<TSchema extends MapSchema>
 		) as TypedField<TSchema["mapFields"]>;
 	}
 
-	// TODO: when appropriate add setter that delegates to field kind specific setter.
-	// public set(key: FieldKey, content: FlexibleFieldContent<TSchema["mapFields"]>): void {
-	// 	const field = this.get(key);
-	// 	if (field.is(SchemaBuilder.optional(this.schema.mapFields.allowedTypes))) {
-	// 		field.setContent(content);
-	// 	} else {
-	// 		assert(
-	// 			field.is(SchemaBuilder.sequence(this.schema.mapFields.allowedTypes)),
-	// 			"unexpected map field kind",
-	// 		);
-	// 		// TODO: fix merge semantics.
-	// 		field.replaceRange(0, field.length, content as Iterable<ContextuallyTypedNodeData>);
-	// 	}
-	// }
+	public set(
+		key: FieldKey,
+		content: FlexibleFieldContent<TSchema["mapFields"]> | undefined,
+	): void {
+		const field = this.getBoxed(key);
+		const fieldSchema = this.schema.mapFields;
+
+		if (fieldSchema.kind === FieldKinds.optional) {
+			const optionalField = field as OptionalField<AllowedTypes>;
+			optionalField.content = content;
+		} else {
+			assert(fieldSchema.kind === FieldKinds.sequence, "Unexpected map field kind");
+
+			const sequenceField = field as Sequence<AllowedTypes>;
+
+			// TODO: fix merge semantics.
```

Comment:
@noencke The more I think about this, the more I'm leaning towards just throwing in this case for now. Since the merge semantics here are questionable, and we don't expect this path to be hittable via our public API, it seems preferable.

Reply:
That makes sense to me. I agree.